### PR TITLE
Theme filter

### PR DESCRIPTION
**Pull request by deekshithmd**
Added dark-theme, test user and  tried to implement data persistence
Main files to check
1. ThemeContext
2. DataContext
3. Login

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1 @@
-REACT_APP_JWT_SECRET=deekshithmogra
\ No newline at end of file
+REACT_APP_JWT_SECRET=deekshithmograecommerce
```
Comment (saurabhchirde) [inline .env:1]: Remove the env file, as it should not be uploaded on Github. You can refer to the below article for help https://daily-dev-tips.com/posts/removing-a-env-file-from-git-history/

File: src/contexts/ThemeContext/ThemeContext.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const Toggle = () => {
+    if (theme === "dark") {
+      setTheme("light");
+      document.body.classList.remove("dark-theme");
+      document.body.classList.add("light-theme");
+      localStorage.setItem("theme", "light");
+    } else {
+      setTheme("dark");
+      document.body.classList.remove("light-theme");
+      document.body.classList.add("dark-theme");
```
Comment (saurabhchirde) [inline src/contexts/ThemeContext/ThemeContext.jsx:18]: For dark mode you can look for the better approach

**Review by saurabhchirde — APPROVED** [review]
Clean, and easy to read code, gave some suggestions else you can proceed to merge.

File: .env (hunk 1 above)
Comment (deekshithmd) [reply .env:1]: ok, I will check that.

File: src/contexts/ThemeContext/ThemeContext.jsx (hunk 2 above)
Comment (deekshithmd) [reply src/contexts/ThemeContext/ThemeContext.jsx:18]: Ok, I will check that

File: src/contexts/DataContext/DataContext.jsx [hunk 3]
```diff
@@ -21,6 +28,9 @@ const DataProvider = ({ children }) => {
     leafy: false,
     marrow: false,
     root: false,
+    alltime: false,
```
Comment (sanjitsarkar) [inline src/contexts/DataContext/DataContext.jsx:31]: You can put this properties leafy,marrow and root into a separate object

File: src/contexts/DataContext/DataContext.jsx [hunk 4]
```diff
@@ -87,28 +110,45 @@ const DataProvider = ({ children }) => {
...
+    : categoryfiltered;
+
+  const ratingfiltered =
+    data.rating === 0
+      ? seasonFiltered
+      : seasonFiltered.filter((item) => item.rating > data.rating);
 
   function getSorted(product, sortBy) {
     const output =
       sortBy === null
```
Comment (sanjitsarkar) [inline src/contexts/DataContext/DataContext.jsx:140]: don't use ternary operator for multi-line statements as it reduces the code readability

File: src/pages/authentication/Login.jsx [hunk 5]
```diff
@@ -11,6 +11,28 @@ export default function Login() {
...
+        localStorage.setItem(
+          "login",
+          JSON.stringify(response.data.encodedToken)
+        );
+        setToken(true);
+        navigate("/");
+      }
+    }
+    catch(e){
+      setError(true)
```
Comment (sanjitsarkar) [inline src/pages/authentication/Login.jsx]: Store the error message in the variable

File: src/pages/authentication/Login.jsx [hunk 6]
```diff
@@ -77,10 +99,13 @@ export default function Login() {
             </section>
             <input
               type="submit"
-              className="btn btn-solid-primary margin"
+              className="btn btn-solid-primary auth-btn margin margiin-l-3-5"
               value="Login"
             />
           </form>
+          <button className="btn btn-solid-primary auth-btn margin" onClick={()=>TestLogin()}>
```
Comment (sanjitsarkar) [inline src/pages/authentication/Login.jsx]: always name the methods/function in camelCase

**Review by sanjitsarkar — APPROVED** [review]

File: src/contexts/Reducers/DataReducer.jsx [hunk 7]
```diff
@@ -38,6 +46,9 @@ export const DataReducer = (state, action) => {
         leafy: false,
         marrow: false,
         root: false,
+        alltime: false,
+        summer: false,
+        winter: false,
       };
     case "LOGOUT":
       return { ...state, wishlist: [], cart: [] };
```
Comment (mayuresh-tech) [inline src/contexts/Reducers/DataReducer.jsx:54]: While logging out you can clean local storage also.
(Don't know if you have handled this but you can keep this as a suggestion)

**Review by mayuresh-tech — APPROVED** [review]
Ok to merge!

File: src/contexts/DataContext/DataContext.jsx (hunk 3 above)
Comment (deekshithmd) [reply src/contexts/DataContext/DataContext.jsx:31]: Ok, I will look into this.

File: src/pages/authentication/Login.jsx (hunk 6 above)
Comment (deekshithmd) [reply src/pages/authentication/Login.jsx]: ok, Thank you for review

File: src/contexts/Reducers/DataReducer.jsx (hunk 7 above)
Comment (deekshithmd) [reply src/contexts/Reducers/DataReducer.jsx:54]: yes, did that in handlelogout function

**Pull request merged**